src/stateMachines/featuresConfiguration.ts

```diff
@@ -0,0 +1,305 @@
+import { AMT, CIM, Common, IPS } from '@open-amt-cloud-toolkit/wsman-messages'
+import { HttpHandler } from '../HttpHandler'
+import Logger from '../Logger'
+import { assign, createMachine, interpret } from 'xstate'
+import { AMTConfiguration, AMTRedirectionServiceEnabledStates, mapAMTUserConsent } from '../models'
+import { devices } from '../WebSocketListener'
+import { ClientResponseMsg } from '../utils/ClientResponseMsg'
+import { parseBody } from '../utils/parseWSManResponseBody'
+import { AMT_REDIRECTION_SERVICE_ENABLE_STATE } from '@open-amt-cloud-toolkit/wsman-messages/models/common'
+import { RedirectionService } from '@open-amt-cloud-toolkit/wsman-messages/amt/models'
+import { IPS_OptInService } from '@open-amt-cloud-toolkit/wsman-messages/models/ips_models'
+
+export class FeaturesConfiguration {
+  amt: AMT.Messages
+  cim: CIM.Messages
+  ips: IPS.Messages
+  // the class name and method for this is silly
+  // should be new ClientReponseMsg(params...)
+  // but instead it has a method called .get?
+  clientMsgBuilder: ClientResponseMsg
+  httpHandler: HttpHandler
+  logger: Logger
+
+  constructor (
+    private readonly clientId: string,
+    private readonly amtCfg: AMTConfiguration
```

Comment:
nit: stick with amtConfig, instead cfg for shortening configuration, here and other places